packages/deployments/contracts/contracts/lib/Connext/ConnextUtils.sol

```diff
@@ -49,13 +49,38 @@ library ConnextUtils {
     return keccak256(abi.encode(transfer));
   }
 
+  function verifyRouterPermit(
+    bytes32 _permit,
+    address[] calldata _routers,
+    bytes[] calldata _signatures
+  ) external view {
+    for (uint i = 0; i < _signatures.length; i++) {
+      // TODO: If the sender *is* the router, skip
+      if (msg.sender == _routers[i]) {
+        continue;
+      }
+      // TODO: Use Error type, revert
+      require(recoverHashedSignature(_permit, _signatures[i]) == _routers[i], "Invalid signature");
+    }
+  }
+
+  /**
+   * @notice Holds the logic to recover the signer from a hashed encoded payload.
+   * @param _hashed - The bytes32 payload that was signed
+   * @param _sig - The signature you are recovering the signer from
+   */
+  function recoverHashedSignature(bytes32 _hashed, bytes calldata _sig) public pure returns (address) {
```

Comment:
why not pass in the encoded information instead of the hash of it and always use the same recovery method? we should only have one of these low-level recovery functions

Reply:
Yeah I agree. However, I was trying to do the gas-efficient thing of recycling calldata, and reduce the number of `abi.encode` and `keccak256` calls here, since we might be looping over multiple router sigs. I'll see what I can do to reduce this to one function